WombatProject/Controllers/WishlistsController.cs

```diff
@@ -43,66 +40,28 @@ public async Task<ActionResult<Wishlist>> GetWishlist(int id)
             return wishlist;
         }
 
-        // PUT: api/Wishlists/5
-        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutWishlist(int id, Wishlist wishlist)
-        {
-            if (id != wishlist.Id)
-            {
-                return BadRequest();
-            }
-
-            _apiService.Context.Entry(wishlist).State = EntityState.Modified;
-
-            try
-            {
-                await _apiService.Context.SaveChangesAsync();
-            }
-            catch (DbUpdateConcurrencyException)
-            {
-                if (!WishlistExists(id))
-                {
-                    return NotFound();
-                }
-                else
-                {
-                    throw;
-                }
-            }
-
-            return NoContent();
-        }
-
         // POST: api/Wishlists
-        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
         [HttpPost]
-        public async Task<ActionResult<Wishlist>> PostWishlist(Book book)
+        public async Task<ActionResult<Wishlist>> AddItemToWishlist(Book book)
         {
-            var wishlist = await _apiService.AddBookToWishlist(book);
+            var wishlist = await _repository.AddBookToWishlistAsync(book);
 
-            return CreatedAtAction("GetWishlist", new { id = wishlist.Id }, wishlist);
+            return CreatedAtAction("GetWishlistItemById", new { id = wishlist.Id }, wishlist);
         }
 
         // DELETE: api/Wishlists/5
         [HttpDelete("{id}")]
-        public async Task<IActionResult> DeleteWishlist(int id)
+        public async Task<IActionResult> RemoveBookFromWishlist(int id)
```

Comment:
Method name can be: RemoveBookFromBookshelfById

Reply:
Consider it done! :100: